taichi/ui/backends/vulkan/app_context.h

```diff
@@ -37,13 +37,36 @@ class TI_DLL_EXPORT AppContext {
 
   AppConfig config;
 
+  taichi::lang::Pipeline *get_raster_pipeline(
+      const std::string &frag_path,
+      const std::string &vert_path,
+      taichi::lang::TopologyType prim_topology,
+      bool depth = false,
+      taichi::lang::PolygonMode polygon_mode = taichi::lang::PolygonMode::Fill,
+      bool blend = true,
+      bool vbo_instanced = false);
+
+  taichi::lang::Pipeline *get_customized_raster_pipeline(
```

Comment:
what's the different between these two methods?

Reply:
is this about vertex normal/color support?

Reply:
yes, and currently used for line drawing because we use a compute shader to generate translated vertices and that vertex buffer is slimmer to save space

Reply:
will provide comments